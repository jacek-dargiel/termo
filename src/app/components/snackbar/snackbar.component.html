<aside class="mdc-snackbar" #snackbarRef>
  <div class="mdc-snackbar__surface" role="status" aria-relevant="additions">
    <div class="mdc-snackbar__label" aria-atomic="false"></div>
  </div>
</aside>
