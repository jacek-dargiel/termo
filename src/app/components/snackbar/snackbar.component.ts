import { Component, AfterViewInit, ViewChild, ElementRef, OnDestroy } from '@angular/core';
import { MDCSnackbar } from '@material/snackbar';
import { MDCSnackbarData } from '@material/snackbar/foundation';

import { SnackbarService } from '../../services/snackbar.service';

@Component({
  selector: 'termo-snackbar',
  templateUrl: './snackbar.component.html',
  styleUrls: ['./snackbar.component.scss']
})
export class SnackbarComponent implements AfterViewInit, OnDestroy {
  snackbar: MDCSnackbar;
  messagesSub = this.snackbarService.messages
    .subscribe((dataObject: MDCSnackbarData) => this.showSnackbar(dataObject));
  @ViewChild('snackbarRef') el: ElementRef<HTMLElement>;

  constructor(
    private snackbarService: SnackbarService,
  ) { }

  ngAfterViewInit() {
    this.snackbar = new MDCSnackbar(this.el.nativeElement);
  }

  ngOnDestroy() {
    this.messagesSub.unsubscribe();
  }

  showSnackbar(dataObject: MDCSnackbarData) {
    if (!this.snackbar) {
      console.error('Can\'t show snackbar before view init.', dataObject);
      return;
    }
    this.snackbar.show(dataObject);
  }

}
