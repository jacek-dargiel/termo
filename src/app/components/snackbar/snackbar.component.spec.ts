import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { SnackbarComponent } from './snackbar.component';
import { Subject } from 'rxjs';
import { SnackbarService } from '../../services/snackbar.service';

class MockSnapshotService {
  messages = new Subject();
}

describe('SnackbarComponent', () => {
  let component: SnackbarComponent;
  let fixture: ComponentFixture<SnackbarComponent>;
  let service: MockSnapshotService;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ SnackbarComponent ],
      providers: [
        { provide: SnackbarService, useClass: MockSnapshotService }
      ],
    })
      .compileComponents();

    service = TestBed.get(SnackbarService);
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SnackbarComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render the snackbar component', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should render a snackbar when message recived', () => {
    let message = {
      message: 'Something is wrong',
    };
    service.messages.next(message);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
