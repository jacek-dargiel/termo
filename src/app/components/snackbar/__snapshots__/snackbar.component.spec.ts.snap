// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnackbarComponent should render a snackbar when message recived 1`] = `
<termo-snackbar
  el={[Function ElementRef]}
  messagesSub={[Function Subscriber]}
  snackbar={[Function MDCSnackbar]}
  snackbarService={[Function MockSnapshotService]}
>
  <div
    aria-atomic="true"
    aria-live="assertive"
    class="mdc-snackbar mdc-snackbar--align-start mdc-snackbar--active"
  >
    <div
      class="mdc-snackbar__text"
    >
      Something is wrong
    </div>
    <div
      class="mdc-snackbar__action-wrapper"
    >
      <button
        aria-hidden="true"
        class="mdc-snackbar__action-button"
        type="button"
      />
    </div>
  </div>
</termo-snackbar>
`;

exports[`SnackbarComponent should render the snackbar component 1`] = `
<termo-snackbar
  el={[Function ElementRef]}
  messagesSub={[Function Subscriber]}
  snackbar={[Function MDCSnackbar]}
  snackbarService={[Function MockSnapshotService]}
>
  <div
    aria-atomic="true"
    aria-hidden="true"
    aria-live="assertive"
    class="mdc-snackbar mdc-snackbar--align-start"
  >
    <div
      class="mdc-snackbar__text"
    />
    <div
      class="mdc-snackbar__action-wrapper"
    >
      <button
        aria-hidden="true"
        class="mdc-snackbar__action-button"
        type="button"
      />
    </div>
  </div>
</termo-snackbar>
`;
